File: SMSnmdsathletic/SMSnmdsathletic.m
% clear variables and close windows
clear all
close all
clc

% load data
load('athletic.mat')

x = athletic.data; % subsample
x = zscore(x);     % standardized
d = dist(x');      % distance matrix

% nonmetric MDS
[Y,stress,disparities] = mdscale(d,2,'criterion','stress');

% plot
rows = athletic.textdata(2:end,1);
plot(Y(:,1),-Y(:,2),'wo')
title('athletic records','FontSize',16,'FontWeight','Bold')
ylabel('x_2','FontSize',16,'FontWeight','Bold')
xlabel('x_1','FontSize',16,'FontWeight','Bold')
text(Y(:,1),-Y(:,2),rows,'FontSize',14)
set(gca,'LineWidth',1.6,'FontSize',16,'FontWeight','Bold')

File: SMSnmdsathletic/SMSnmdsathletic.R
# clear variables and close windows
rm(list=ls(all=TRUE))
graphics.off()

# install and load packages
libraries = c("MASS")
lapply(libraries, function(x) if (!(x %in% installed.packages())) {
    install.packages(x)})
lapply(libraries, library, quietly = TRUE, character.only = TRUE)

# setwd("C:/...") # please set the working directory
# load data
load("athletic.rda")

# compute distance
athletic.dist = dist(scale(athletic))

# nonmetric MDS
athletic.mds = isoMDS(athletic.dist,maxit=1000,tol = 1e-6)
xylim        = range(athletic.mds$points[,1],-athletic.mds$points[,2])

# plot
plot(athletic.mds$points[,1],-athletic.mds$points[,2], main="athletic records", type = "n", ylab=expression(x[2]), xlab=expression(x[1]),xlim=xylim,ylim=xylim)
text(athletic.mds$points[,1],-athletic.mds$points[,2], labels = row.names(athletic),xpd=NA)